### PR TITLE
Add error and warning message for kit zip code rule

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

Allow configuring templates with a kit `ZIP_CODE` rule, so studies can configure what error/warning message to show when user provided zip code is not in supported list.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [x] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [x] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/BuilderUtils.java [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
 import org.broadinstitute.ddp.studybuilder.task.CustomTask;
+import org.broadinstitute.ddp.util.ConfigUtil;
+import org.broadinstitute.ddp.util.GsonPojoValidator;
+import org.broadinstitute.ddp.util.GsonUtil;
+import org.broadinstitute.ddp.util.JsonValidationError;
 
 /**
  * Various helpers for study-builder.
  */
 class BuilderUtils {
```
Comment (ssettipalli) [inline study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/BuilderUtils.java:18]: Looks like its all Template related helpers, can we name it TemplateBuilderUtils ?

Comment (yufengwng) [reply study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/BuilderUtils.java:18]: There's other stuff here too, so I'll keep the name as-is.

Comment (ssettipalli) [reply study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/BuilderUtils.java:18]: May be we should have named StudyBuilderUtils.java

**Review by ssettipalli — APPROVED** [review]
looks good to me

**Pull request merged**